src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -1216,6 +1217,76 @@ static void M1()
             await VerifyAnalyzerCSAsync(csSource);
         }
 
+        [Fact]
+        public async Task CallGuardFromPlatformSpecificAssembly()
+        {
+            string csDependencyCode = @"
+public class Library
+{
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""windows"")]
+    [System.Runtime.Versioning.SupportedOSPlatformGuard(""linux"")]
+    public static bool IsSupported => false;
+
+    public static void AMethod() { }
+}";
+            csDependencyCode = @$"[assembly: System.Runtime.Versioning.SupportedOSPlatform(""windows"")]
+                                  [assembly: System.Runtime.Versioning.SupportedOSPlatform(""linux"")]
+                                  {csDependencyCode}";
+
+            string csCurrentAssemblyCode = @"
+using System;
+
+public class Program
+{
+    public void ProgramMethod()
+    {
+        [|Library.AMethod()|]; // Not guarded, warns
```

Comment:
Can you add a test case where it _is_ guarded?

Reply:
That is just below:
```cs
        if (Library.IsSupported) // this is the guard
        {
             Library.AMethod();
        }
```